test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/ComplexTypeModelBinderProviderTest.cs

```diff
@@ -55,11 +55,81 @@ public void Create_ForSupportedTypes_ReturnsBinder()
             Assert.IsType<ComplexTypeModelBinder>(result);
         }
 
+        [Theory]
+        [InlineData(typeof(StructWithNoDefaultConstructor))]
+        [InlineData(typeof(ClassWithNoDefaultConstructor))]
+        [InlineData(typeof(ClassWithStaticConstructor))]
+        public void Create_ForModelTypeWith_NoDefaultPublicConstructor_ReturnsNull(Type modelType)
+        {
+            // Arrange
+            var provider = new ComplexTypeModelBinderProvider();
+            var context = new TestModelBinderProviderContext(modelType);
+
+            // Act
+            var result = provider.GetBinder(context);
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void Create_ForAbstractModelTypeWith_DefaultPublicConstructor_ReturnsNull()
+        {
+            // Arrange
+            var provider = new ComplexTypeModelBinderProvider();
+            var context = new TestModelBinderProviderContext(typeof(AbstractClassWithDefaultConstructor));
+
+            // Act
+            var result = provider.GetBinder(context);
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        // Actually c# doesn't allow creating parameterless constructors for structs.
```

Comment:
The name of this `struct` is incorrect. Actually C# _always_ defines a parameterless constructor for `struct`s and developers can't override that one. Suggest debugging to see exactly how `ModelMetadata` describes this type to the `ComplexTypeModelBinderProvider`.